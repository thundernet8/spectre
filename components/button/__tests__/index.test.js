import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import Button from "..";

const ButtonGroup = Button.Group;

describe("Button", () => {
    it("should render Button", () => {
        const btn = shallow(<Button />);

        expect(btn.type()).toEqual("button");

        expect(btn.hasClass("spt-btn")).toBe(true);

        const largeBtn = shallow(<Button size="large" type="primary" />);

        expect(largeBtn.hasClass("spt-btn-lg")).toBe(true);
        expect(largeBtn.hasClass("spt-btn-primary")).toBe(true);

        // TODO add icon test
    });

    it("button renders correctly", () => {
        const tree = renderer
            .create(<Button size="large" type="primary" />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe("Button Group", () => {
    it("should render Button Group", () => {
        const group = shallow(<ButtonGroup size="block" />);

        expect(group.type()).toEqual("div");

        expect(group.hasClass("spt-btn-group")).toBe(true);

        expect(group.hasClass("spt-btn-group-block")).toBe(true);
    });

    it("button group renders correctly", () => {
        const tree = renderer.create(<ButtonGroup size="large" />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
