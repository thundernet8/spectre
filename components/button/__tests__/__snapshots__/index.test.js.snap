// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Group button group renders correctly 1`] = `
<div
  className="spt-btn-group spt-btn-group-lg"
/>
`;

exports[`Button button renders correctly 1`] = `
<button
  className="spt-btn spt-btn-primary spt-btn-lg"
  onClick={[Function]}
  onMouseUp={[Function]}
  type="button"
/>
`;
